### PR TITLE
fix: better default size policy for qcollapsible

### LINKED ISSUE #214: Extra Space in Closed QCollabsible
**Describe the bug**
I am seeing a lot of empty space below the QCollabsible for some reason. Sometimes, this depends on what is inside the Qcollabsible. Is this by design?

image

**To Reproduce**
```python
"""Example for QCollapsible"""
from qtpy.QtWidgets import QApplication, QLabel, QPushButton, QWidget, QVBoxLayout, QTableWidget

from superqt import QCollapsible

app = QApplication([])

main_widget = QWidget()
layout = QVBoxLayout()
main_widget.setLayout(layout)


collapsible_bad = QCollapsible("Advanced analysis")
table = QTableWidget(20, 2)
collapsible_bad.addWidget(table)
layout.addWidget(collapsible_bad)


collapsible = QCollapsible("Advanced analysis (not bad)")
collapsible.addWidget(QLabel("This is the inside of the collapsible frame"))
for i in range(3):
    collapsible.addWidget(QPushButton(f"Content button {i + 1}"))
layout.addWidget(collapsible)

layout.addStretch()
main_widget.show()

app.exec_()
```
**Expected behavior**
I would have expected to the space to be minimum between the two widgets in the example above.

**Desktop (please complete the following information):**
 - OS: Windows 11 and Windows 10
 - Qt Backend: PyQt5
 - Python version 3.10

### PR DESCRIPTION
**Pull request by tlambert03**
closes #214

### CODE REVIEW

File: src/superqt/collapsible/_collapsible.py [hunk 1]
```diff
@@ -41,13 +41,15 @@ def __init__(
...
         self._toggle_btn.setStyleSheet("text-align: left; border: none; outline: none;")
         self._toggle_btn.toggled.connect(self._toggle)
 
         # frame layout
-        self.setLayout(QVBoxLayout())
-        self.layout().setAlignment(Qt.AlignmentFlag.AlignTop)
-        self.layout().addWidget(self._toggle_btn)
+        layout = QVBoxLayout(self)
+        layout.setAlignment(Qt.AlignmentFlag.AlignTop)
+        layout.addWidget(self._toggle_btn)
```
Comment (goanpeca) [inline src/superqt/collapsible/_collapsible.py:53]: `self.setLayout(layout)` ?

Comment (tlambert03) [reply src/superqt/collapsible/_collapsible.py:53]: QVBoxLayout(self) has the same effect

Comment (goanpeca) [reply src/superqt/collapsible/_collapsible.py:53]: Ohhh TIL :)

Comment (tlambert03) [reply src/superqt/collapsible/_collapsible.py:53]: I only learned that recently myself! 😃 and I've come to prefer it.

**Pull request merged**